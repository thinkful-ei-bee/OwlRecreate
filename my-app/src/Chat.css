.chat-item{
    display:block;
    text-align:left;
}

.chat-info{
    width: 252px;
    display: inline-block;
    margin-left: 10px;
}

.user-avatar{
    margin-top: 5px;
    vertical-align: top;
    width: 30px;
    height: 30px;
    background-color: #eeeeef;
}

.name{
    font-size: 14.5px;
    display: inline-block;
}

.time{
    font-family: Proxima Nova W01,Avenir Next W01,Avenir Next,sans-serif;
    font-weight: 400;
    font-style: normal;
    color: #8e939a;
    margin-left: 5px;
    display:inline-block;
}


