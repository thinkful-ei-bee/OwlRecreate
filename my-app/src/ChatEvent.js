import React from 'react'

function ChatEvent(props){
  console.log(props.eventData.type,'test chatevent')
  return(
    <div className='chat-event'>
    {
      props.eventData.type==='message'?
      <div className='message-event'>
      <img alt = {props.name}src={props.avatar}></img>
      {props.name} {props.eventData.type}</div>:
      props.eventData.type==='thumb-up'?
      <div className='thumb-up'>{props.name} {props.eventData.type}</div>:
      props.eventData.type==='thumb-down'?
      <div className='thumb-down'>{props.name} {props.eventData.type}</div>:  
      props.eventData.type==='clap'?
      <div className='clap'>{props.name} {props.eventData.type}</div>:
      props.eventData.type==='leave'?
      <div className='leave'>{props.name} {props.eventData.type}</div>:
      props.eventData.type==='join'?
      <div className='join'>{props.name} {props.eventData.type}</div>:
      props.eventData.type==='join-stage'?
      <div className='join-stage'>{props.name} {props.eventData.type}</div>:
      <div className='leave-stage'>{props.name} {props.eventData.type}</div>
  }

    </div>

  )

}

export default ChatEvent