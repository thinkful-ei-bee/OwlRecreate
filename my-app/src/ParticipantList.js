import React from 'react'
import Participant from './Participant'


function ParticipantList(props){
  console.log(props.participantsData)
  const array1=props.participantsData.filter(participant=>participant.onStage)
  const array2=props.participantsData.filter(participant=>participant.inSession && !participant.onStage)
  const array3=props.participantsData.filter(participant=>!participant.onStage&&!participant.inSession)
  const participantData = array1.concat(array2).concat(array3)
  return(
  <div className = 'participants-list'>
    {participantData.map(participant =>
      <Participant key = {participant.id}
          name = {participant.name}
          avatar = {participant.avatar}
          inSession = {participant.inSession}
          onStage ={participant.onStage}
         />)}
  </div>
)
}

export default ParticipantList



          