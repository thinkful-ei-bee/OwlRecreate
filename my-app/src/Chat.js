import React from 'react'
import ChatEvent from './ChatEvent';




function Chat(props){
  
  // restructuring 
  let {chatData,participantsData}=props
  
  
  return(
    <div>
        {
          chatData.map((event,index)=>
            
            {
              
              let participantName = participantsData.find(participant=>participant.id===event.participantId)
              
              
              // because it has more than two items, so return is necessary here. 
             return <ChatEvent eventData={event}
                    name = {participantName.name}
                    key = {'chat-'+index}
              />
            }
            )
            
       
        }
    </div>
  )
}

export default Chat