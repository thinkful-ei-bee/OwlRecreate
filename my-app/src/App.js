import React, { Component } from 'react';
import participants from './ParticipantsData';
import ParticipantList from './ParticipantList';
import chatEvents from './ChatEvents';
import ChatList from './ChatList';
import OnStage from './onStage';
import './App.css';

class App extends Component {
  
  
  render() {
     
    return (
      <div className="App">
        
          <ParticipantList participantsData = {participants}/>

           <OnStage participantsData = {participants}/>
          
          <ChatList chatMessages ={chatEvents}/>
   
        
      </div>
    );
  }
}


export default App;
