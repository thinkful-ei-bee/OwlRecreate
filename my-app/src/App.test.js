import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import participants from './ParticipantsData';
import ParticipantList from './ParticipantList';
import renderer from 'react-test-renderer';
import Chat from './Chat';
import chatEvents from './ChatData'
import ChatEvent from './ChatEvent'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

//chat
it('chat renders the UI as expected',()=>{
  const tree = renderer.create(<Chat participantsData = {participants} chatData={chatEvents}/>).toJSON();
  expect(tree).toMatchSnapshot();
});


//part
it('participantList renders the UI as expected',()=>{
  const tree = renderer.create(<ParticipantList participantsData = {participants}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

//chatEvent
it('chatEvent renders the UI as expected',()=>{
  const tree = renderer.create(<ChatEvent eventData={{
    participantId: 4,
    type: 'thumbs-down',
    timestamp: 1548852484247
}} name = {'Frank Runciman'} avatar={'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1'}/>)
.toJSON();
expect(tree).toMatchSnapshot();
})
