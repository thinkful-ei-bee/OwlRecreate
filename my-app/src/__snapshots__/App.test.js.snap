// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat renders the UI as expected 1`] = `
<div>
  <div
    className="chat-event"
  >
    <div
      className="message-event"
    >
      <img
        alt="Ashla Attwool"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      Ashla Attwool
       
      message
    </div>
  </div>
  <div
    className="chat-event"
  >
    <div
      className="leave-stage"
    >
      Caty Flucker
       
      thumbs-up
    </div>
  </div>
  <div
    className="chat-event"
  >
    <div
      className="leave-stage"
    >
      Frank Runciman
       
      thumbs-down
    </div>
  </div>
  <div
    className="chat-event"
  >
    <div
      className="leave-stage"
    >
      Koren Templeton
       
      raise-hand
    </div>
  </div>
  <div
    className="chat-event"
  >
    <div
      className="clap"
    >
      Koren Templeton
       
      clap
    </div>
  </div>
  <div
    className="chat-event"
  >
    <div
      className="join"
    >
      Koren Templeton
       
      join
    </div>
  </div>
  <div
    className="chat-event"
  >
    <div
      className="leave"
    >
      Axe Kubicka
       
      leave
    </div>
  </div>
  <div
    className="chat-event"
  >
    <div
      className="join-stage"
    >
      Ashla Attwool
       
      join-stage
    </div>
  </div>
  <div
    className="chat-event"
  >
    <div
      className="leave-stage"
    >
      Ashla Attwool
       
      leave-stage
    </div>
  </div>
</div>
`;

exports[`chatEvent renders the UI as expected 1`] = `
<div
  className="chat-event"
>
  <div
    className="leave-stage"
  >
    Frank Runciman
     
    thumbs-down
  </div>
</div>
`;

exports[`participantList renders the UI as expected 1`] = `
<div
  className="participants-list"
>
  <div
    className="session-participant"
  >
    <img
      alt="Koren Templetonavatar"
      className="user-avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="name"
      >
         
        Koren Templeton
         
      </h6>
      <div
        className="participant-status participant-status__green"
      >
        on stage 
      </div>
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Caty Fluckeravatar"
      className="user-avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="name"
      >
         
        Caty Flucker
         
      </h6>
      <div
        className="participant-status participant-status__green"
      >
        on stage 
      </div>
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Ashla Attwoolavatar"
      className="user-avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="name"
      >
         
        Ashla Attwool
         
      </h6>
      <div
        className="participant-status participant-status__green"
      >
        on stage 
      </div>
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Frank Runcimanavatar"
      className="user-avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="name"
      >
         
        Frank Runciman
         
      </h6>
      <div
        className="participant-status participant-status__green"
      >
        in session
      </div>
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Axe Kubickaavatar"
      className="user-avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="name"
      >
         
        Axe Kubicka
         
      </h6>
      <div
        className="participant-status participant-status__gray"
      >
        left session
      </div>
    </div>
  </div>
</div>
`;
